tests/regression/src/lib.rs

```diff
@@ -166,12 +180,18 @@ mod tests {
                 commit_hash: git::extract_commit_hash(&raw_files[1]),
             };
 
-            if git::is_older_commit(&profile1.commit_hash, &profile2.commit_hash) {
-                (profile1, profile2)
-            } else if git::is_older_commit(&profile2.commit_hash, &profile1.commit_hash) {
-                (profile2, profile1)
-            } else {
-                panic!("The commits are not in the same log");
+            match (git::is_mainline(&profile1.commit_hash), git::is_mainline(&profile2.commit_hash)) {
```

Comment:
Rather than spelling out the logic for each individual branch, I think this could be described more succinctly as 
```
If exactly one commit is on main, then that is the baseline. Otherwise, the baseline is the oldest commit.
```